### PR TITLE
Add audio support

### PR DESCRIPTION
**Pull request by emilolosutean**
Support spark collection content republish

### CODE REVIEW

**Comment by emilolosutean** [comment]
[Modify publish-failure-resolver-go to republish Audio](https://trello.com/c/a3WSHHQU/195-modify-publish-failure-resolver-go-to-republish-audio)

File: main.go [hunk 1]
```diff
@@ -22,54 +22,54 @@ const (
...
 	},
 	"next-video-editor": {
-		name:           "video-metadata",
-		originSystemID: "next-video-editor",
-		notifierApp:    cmsMetadataNotifier,
-		scope:          scopeMetadata,
+		name: "video-metadata",
+		defaultOriginSystemID: "http://cmdb.ft.com/systems/next-video-editor",
+		notifierApp:           cmsMetadataNotifier,
+		scope:                 scopeMetadata,
```
Comment (bulicamihai) [inline main.go:72]: Formatting: the names of collections aren't aligned with the other attributes.

File: notifierClient.go [hunk 2]
```diff
@@ -12,8 +12,14 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+const (
+	contentTypeHeader     string = "Content-Type"
+	xOriginSystemIDHeader string = "X-Origin-System-Id"
+	originSystemIDHeader  string = "Origin-System-Id"
+)
```
Comment (bulicamihai) [inline notifierClient.go:19]: In `const` isn't necessary to specify the type

File: uuidCollectionRepublisher_test.go [hunk 3]
```diff
@@ -9,18 +9,29 @@ import (
...
 	republisher := newNotifyingUCRepublisher(mockedNotifierClient, mockedNativeStoreClient, 500*time.Millisecond)
 	collection := targetSystem{
-		name:           "methode",
-		originSystemID: "methode-web-pub",
-		notifierApp:    "cms-notifier",
-		scope:          "content",
+		name: "methode",
+		defaultOriginSystemID: "methode-web-pub",
+		notifierApp:           "cms-notifier",
+		scope:                 "content",
```
Comment (bulicamihai) [inline uuidCollectionRepublisher_test.go:34]: Formatting: the collection's name

File: uuidCollectionRepublisher_test.go [hunk 4]
```diff
@@ -41,14 +53,14 @@ func TestRepublishOk_Ok(t *testing.T) {
...
 	republisher := newNotifyingUCRepublisher(mockedNotifierClient, mockedNativeStoreClient, 1*time.Millisecond)
 	collection := targetSystem{
-		name:           "methode",
-		originSystemID: "methode-web-pub",
-		notifierApp:    "cms-notifier",
-		scope:          "content",
+		name: "methode",
+		defaultOriginSystemID: "methode-web-pub",
+		notifierApp:           "cms-notifier",
+		scope:                 "content",
```
Comment (bulicamihai) [inline uuidCollectionRepublisher_test.go:63]: Formatting: the collection's name

File: uuidCollectionRepublisher_test.go [hunk 5]
```diff
@@ -60,14 +72,14 @@ func TestRepublishNotFound_NotFound(t *testing.T) {
...
 	republisher := newNotifyingUCRepublisher(mockedNotifierClient, mockedNativeStoreClient, time.Second)
 	collection := targetSystem{
-		name:           "methode",
-		originSystemID: "methode-web-pub",
-		notifierApp:    "cms-notifier",
-		scope:          "content",
+		name: "methode",
+		defaultOriginSystemID: "methode-web-pub",
+		notifierApp:           "cms-notifier",
+		scope:                 "content",
```
Comment (bulicamihai) [inline uuidCollectionRepublisher_test.go:82]: Formatting: the collection's name

File: uuidCollectionRepublisher_test.go [hunk 6]
```diff
@@ -83,15 +95,15 @@ func TestRepublishErrNative_Err(t *testing.T) {
...
 	republisher := newNotifyingUCRepublisher(mockedNotifierClient, mockedNativeStoreClient, 500*time.Millisecond)
 	collection := targetSystem{
-		name:           "methode",
-		originSystemID: "methode-web-pub",
-		notifierApp:    "cms-notifier",
-		scope:          "content",
+		name: "methode",
+		defaultOriginSystemID: "methode-web-pub",
+		notifierApp:           "cms-notifier",
+		scope:                 "content",
```
Comment (bulicamihai) [inline uuidCollectionRepublisher_test.go:106]: Formatting: the collection's name

File: uuidRepublisher_test.go [hunk 7]
```diff
@@ -88,20 +88,21 @@ func TestOkAndSoftErrors_Ok(t *testing.T) {
...
 	mockedUCRepublisher.On("RepublishUUIDFromCollection", "b3ec9282-1073-46ad-9d44-144dad7fe956", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["v1-metadata"]).Return(&msg1, true, nil)
 	mockedUCRepublisher.On("RepublishUUIDFromCollection", "b3ec9282-1073-46ad-9d44-144dad7fe956", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["pac-metadata"]).Return(&msg2, true, nil)
 	mockedUCRepublisher.On("RepublishUUIDFromCollection", "b3ec9282-1073-46ad-9d44-144dad7fe956", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["wordpress"]).Return(nilMsg, false, nil)
 	mockedUCRepublisher.On("RepublishUUIDFromCollection", "b3ec9282-1073-46ad-9d44-144dad7fe956", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["spark"]).Return(nilMsg, false, nil)
+	mockedUCRepublisher.On("RepublishUUIDFromCollection", "b3ec9282-1073-46ad-9d44-144dad7fe956", mock.MatchedBy(func(tid string) bool { return strings.HasPrefix(tid, "prefix1") }), testCollections["spark"]).Return(nilMsg, false, nil)
```
Comment (bulicamihai) [inline uuidRepublisher_test.go]: why this line is duplicated?

File: uuidRepublisher_test.go [hunk 8]
```diff
@@ -11,49 +11,49 @@ import (
...
 var testCollectionsSingle = map[string]targetSystem{
 	"methode": {
-		name:           "methode",
-		originSystemID: "methode-web-pub",
-		notifierApp:    cmsNotifier,
-		scope:          scopeContent,
+		name: "methode",
+		defaultOriginSystemID: "methode-web-pub",
+		notifierApp:           cmsNotifier,
+		scope:                 scopeContent,
```
Comment (bulicamihai) [inline uuidRepublisher_test.go:56]: Formatting: the collection's name at all above

File: main.go (hunk 1 above)
Comment (emilolosutean) [reply main.go:72]: VS Code is doing this formatting, won't change

File: notifierClient.go (hunk 2 above)
Comment (emilolosutean) [reply notifierClient.go:19]: if is not specified, the variable is untyped;  if is typed you cannot assign it to a Go variable of a different type

File: uuidCollectionRepublisher_test.go (hunk 4 above)
Comment (emilolosutean) [reply uuidCollectionRepublisher_test.go:63]: VS Code is doing this formatting, won't change

File: uuidRepublisher_test.go (hunk 7 above)
Comment (emilolosutean) [reply uuidRepublisher_test.go]: checks that repeated publications don't fail

**Review by bulicamihai — APPROVED** [review]

**Pull request merged**